include/boost/geometry/util/range.hpp

```diff
@@ -107,25 +107,13 @@ at(RandomAccessRange & rng,
 \ingroup utility
 */
 template <typename Range>
-inline typename boost::range_value<Range>::type const&
+inline typename boost::range_reference<Range>::type
 front(Range const& rng)
 {
     BOOST_ASSERT(!boost::empty(rng));
     return *boost::begin(rng);
 }
 
-/*!
-\brief Short utility to conveniently return the front element of a Range.
-\ingroup utility
-*/
-template <typename Range>
-inline typename boost::range_value<Range>::type &
-front(Range & rng)
-{
-    BOOST_ASSERT(!boost::empty(rng));
-    return *boost::begin(rng);
-}
-
```

Comment:
Yes that what I mentionned in my second comment. But consider:
```
boost::iterator_range<const int*> r(10, 0);
int & i = bg::range::front(r);
```
That won't compile. I believe this is the reason why Boost.Range does not have this function.
`boost::iterator_range<const point*>` should be valid Linestring, right ?